controllers/profiles/reconciler_dev.go

```diff
@@ -61,41 +65,80 @@ func (d developmentProfile) GetProfile() Profile {
 	return Development
 }
 
-func newDevProfileReconciler(client client.Client, logger *logr.Logger) ProfileReconciler {
+func newDevProfileReconciler(client client.Client, config *rest.Config, logger *logr.Logger) ProfileReconciler {
```

Comment:
why call it newDevProfile, isn't better just devProfileReconciler?

Reply:
It is a code convention in go pre-pending the new for methods that are in charge of creating objects.
In this case we don't want to export the method outside so we are using the lower case.
I used the same style/convention because I've seen that is in line with what @ricardozanini introduced

Reply:
ok, thanks for the explanation.